docs/en/user_guides/1_config.md

```diff
@@ -0,0 +1,166 @@
+  change this key, you may specify `--cfg-options workflow="[(train,1),(val,1)]"`. Note that the quotation mark " is necessary to
+  support list/tuple data types, and that **NO** white space is allowed inside the quotation marks in the specified value.
+
+## Config File Structure
+
+There are 4 basic component types under `config/_base_`, dataset, model, default_metrics, default_runtime.
+Many methods could be easily constructed with one of each like StyleGAN2, CycleGAN, SinGAN.
+Configs consisting of components from `_base_` are called _primitive_.
+
+For all configs under the same folder, it is recommended to have only **one** _primitive_ config. All other configs should inherit from the _primitive_ config. In this way, the maximum of inheritance level is 3.
+
+For easy understanding, we recommend contributors to inherit from existing methods.
+For example, if some modification is made base on StyleGAN2, user may first inherit the basic StyleGAN2 structure by specifying `_base_ = ../styleganv2/stylegan2_c2_ffhq_256_b4x8_800k.py`, then modify the necessary fields in the config files.
+
+If you are building an entirely new method that does not share the structure with any of the existing methods, you may create a folder `xxxgan` under `configs`,
+
+Please refer to [MMEngine](https://github.com/open-mmlab/mmengine/blob/main/docs/en/tutorials/config.md) for detailed documentation.
+
+## Config Name Style
+
+We follow the below style to name config files. Contributors are advised to follow the same style.
+
+```
+{model}_[module setting]_{training schedule}_{dataset}
+```
+
+`{xxx}` is required field and `[yyy]` is optional.
+
+- `{model}`: model type like `stylegan`, `dcgan`, etc. Settings referred in the original paper are included in this field as well (e.g., Stylegan2-config-f)
+- `[module setting]`: specific setting for some modules, including learning rate (e.g., `Glr4e-4_Dlr1e-4` for dcgan), loss terms (`gamma32.8` for stylegan3) and network structures (e.g., `woReLUInplace` in sagan). In this section, information from different submodules (e.g., generator and discriminator) are connected with `_`.
+- `[batch_per_gpu x gpu]`: GPUs and samples per GPU, `b4x8` is used by default in stylegan2.
+- `{schedule}`: training schedule. Including learning rate (e.g., `lr1e-3`), number of gpu and batch size is used (e.g., `8xb32`), and total iterations (e.g., `160kiter`) or number of images shown in the discriminator (e.g., `12Mimgs`).
+- `{dataset}`: dataset like `ffhq`, `lsun-car`, `celeba-hq`.
+
+## An Example of StyleGAN2
+
+To help the users have a basic idea of a complete config and the modules in a modern detection system,
+we make brief comments on the config of Stylegan2 at 256x256 scale.
+For more detailed usage and the corresponding alternative for each module, please refer to the API documentation and the [tutorial in MMDetection](https://github.com/open-mmlab/mmdetection/blob/master/docs/en/tutorials/config.md).
+
+```python
```

Comment:
you can break this long snippet of config into several parts and explain each part briefly, e.g., mode, train_dataloader, test_dataloader, etc.